apps/server/src/modules/h5p-editor/contentStorage/contentStorage.spec.ts

```diff
@@ -1,471 +1,905 @@
+			});
+
+			it('should return content id', async () => {
+				const {
+					newContent: { metadata, content },
+					user,
+				} = setup();
+				contentRepo.save.mockImplementationOnce(helpers.repoSaveMock);
+
+				const id = await service.addContent(metadata, content, user);
+
+				expect(id).toBeDefined();
 			});
 		});
-		describe('WHEN content can not be added error', () => {
-			it('should throw new error', async () => {
-				const { metadata, content, user } = setup();
-				jest.spyOn(fsPromiseMock, 'writeFile').mockImplementationOnce(() => {
-					throw new Error('Could not write file');
-				});
-				try {
-					await service.addContent(metadata, content, user);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+
+		describe('WHEN modifying existing content', () => {
+			it('should call H5pContentRepo.findById', async () => {
+				const {
+					existingContent: { metadata, content, id },
+					user,
+				} = setup();
+
+				await service.addContent(metadata, content, user, id);
+
+				expect(contentRepo.findById).toHaveBeenCalledWith(id);
+			});
+
+			it('should call H5pContentRepo.save', async () => {
+				const {
+					existingContent,
+					newContent: { metadata, content },
+					user,
+				} = setup();
+				contentRepo.findById.mockResolvedValueOnce(existingContent);
+
+				await service.addContent(metadata, content, user, existingContent.id);
+
+				expect(contentRepo.save).toHaveBeenCalledWith(expect.objectContaining({ metadata, content }));
+			});
+
+			it('should save content and return existing content id', async () => {
+				const {
+					existingContent,
+					newContent: { metadata, content },
+					user,
+				} = setup();
+				const oldId = existingContent.id;
+				contentRepo.save.mockImplementationOnce(helpers.repoSaveMock);
+				contentRepo.findById.mockResolvedValueOnce(existingContent);
+
+				const newId = await service.addContent(metadata, content, user, oldId);
+
+				expect(newId).toEqual(oldId);
+				expect(existingContent).toEqual(expect.objectContaining({ metadata, content }));
 			});
 		});
-		describe('WHEN contentId can not be generated', () => {
-			it('should throw new error', async () => {
-				const { metadata, content, user } = setup();
-				// eslint-disable-next-line @typescript-eslint/require-await
-				jest.spyOn(fsPromiseMock, 'access').mockImplementation(async () => undefined);
-				try {
-					await service.addContent(metadata, content, user);
-				} catch (err) {
-					expect(err).toBeInstanceOf(Error);
-				}
+
+		describe('WHEN saving content fails', () => {
+			it('should throw an InternalServerErrorException', async () => {
+				const {
+					existingContent: { metadata, content },
+					user,
+				} = setup();
+				contentRepo.save.mockRejectedValueOnce(new Error());
+
+				const addContentPromise = service.addContent(metadata, content, user);
+
+				await expect(addContentPromise).rejects.toBeInstanceOf(InternalServerErrorException);
```

Comment:
```suggestion
				await expect(addContentPromise).rejects.toThrow(InternalServerErrorException);
```
Apparently it makes no difference for the outcome, but I think "toThrow" is a bit more expressive. And above you also say "it('should throw an InternalServerErrorException'"

Reply:
Changed for all instances in entire file